docker_build/Dockerfile

```diff
@@ -0,0 +1,80 @@
+# An integration test & dev container which builds and installs cuDF from main
+ARG CUDA_VERSION=10.2
+ARG CUDA_SHORT_VERSION=${CUDA_VERSION}
+
+FROM nvidia/cuda:${CUDA_VERSION}-devel
+ENV LD_LIBRARY_PATH=$LD_LIBRARY_PATH:/usr/local/cuda/lib64:/usr/local/lib
+# Needed for cudf.concat(), avoids "OSError: library nvvm not found"
+ENV NUMBAPRO_NVVM=/usr/local/cuda/nvvm/lib64/libnvvm.so
+ENV NUMBAPRO_LIBDEVICE=/usr/local/cuda/nvvm/libdevice/
+ENV DEBIAN_FRONTEND=noninteractive
+
+ENV CC=/usr/bin/gcc
+ENV CXX=/usr/bin/g++
+
+# Build env variables for arrow
+ENV CMAKE_BUILD_TYPE=release
+ENV PYARROW_WITH_PARQUET=1
+ENV PYARROW_WITH_CUDA=1
+ENV PYARROW_WITH_ORC=1
+ENV PYARROW_WITH_DATASET=1
+
+ENV ARROW_HOME=/repos/dist
+ENV LD_LIBRARY_PATH=/repos/dist/lib:$LD_LIBRARY_PATH
+
+# Build env variables for rmm
+ENV INSTALL_PREFIX=/usr
+
+
+RUN apt update -y --fix-missing && \
+    apt upgrade -y && \
+    apt install -y --no-install-recommends \
+      git \
+      libboost-all-dev \
+      python3.7-dev \
+      build-essential \
+      cmake \
+      software-properties-common \
+      llvm-7 \
+      autoconf \
+      bison \
+      flex \
+      libboost-filesystem-dev \
+      libboost-system-dev \
+      libboost-regex-dev \
+      libjemalloc-dev \
+      wget \
+      libssl-dev \
+      protobuf-compiler \
+      clang && \
+    apt-get autoremove -y && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/* && \
+    add-apt-repository ppa:deadsnakes/ppa && \
```

Comment:
What is this ppa used for?

Reply:
This was to get 3.7 into a 16.04 install, the changes above pull a `nvidia/cuda:11.2.2-devel-ubuntu20.04` image so this is probably not needed

Reply:
I have check all version of cuda devel & devel-ubuntu20.04 images, it seems python isn't installed out of the box, so we would need this to install python for all images.